_posts/en/2024-10-09-wallet-integration-guide.md

```diff
@@ -0,0 +1,590 @@
+
+# Sign and send transaction with P2A output
+bitcoin-cli -regtest -rpcwallet=test signrawtransactionwithwallet 02000000013b64742e658ff39da0288a5010f094a9e94e6ce3a3ce2ed84b74babc017aea490000000000fdffffff02260100000000000016001400991cdadccdf30cb5a04663b0371cb433a095b4f0000000000000000451024e7300000000
+{
+  "hex": "020000000001013b64742e658ff39da0288a5010f094a9e94e6ce3a3ce2ed84b74babc017aea490000000000fdffffff02260100000000000016001400991cdadccdf30cb5a04663b0371cb433a095b4f0000000000000000451024e7302473044022002c7e756b15135a3c0a061df893a857b42572fd816e41d3768511437baaeee4102200c51fcce1e5afd69a28c2d48a74fd5e58b280b7aa2f967460673f6959ab565e80121030af1fadce80bcb8ba614634bc82c71eea2ed87a5692d3127766cc896cef1bdb100000000",
+  "complete": true
+
+# Turns off sanity fee check
+bitcoin-cli -regtest -rpcwallet=test sendrawtransaction 020000000001013b64742e658ff39da0288a5010f094a9e94e6ce3a3ce2ed84b74babc017aea490000000000fdffffff02260100000000000016001400991cdadccdf30cb5a04663b0371cb433a095b4f0000000000000000451024e7302473044022002c7e756b15135a3c0a061df893a857b42572fd816e41d3768511437baaeee4102200c51fcce1e5afd69a28c2d48a74fd5e58b280b7aa2f967460673f6959ab565e80121030af1fadce80bcb8ba614634bc82c71eea2ed87a5692d3127766cc896cef1bdb100000000 "0"
+fdee3b6a5354f31ce32242db10eb9ee66017e939ea87db0c39332262a41a424b
+
+# Replaced prior package
+bitcoin-cli -regtest getrawmempool
+[
+  "fdee3b6a5354f31ce32242db10eb9ee66017e939ea87db0c39332262a41a424b"
+]
+
+# For child, burn value to fees, make 65vbyte tx with OP_RETURN to avoid tx-size-small error
+bitcoin-cli -regtest createrawtransaction '[{"txid": "fdee3b6a5354f31ce32242db10eb9ee66017e939ea87db0c39332262a41a424b", "vout": 1}]' '[{"data": "feeeee"}]'
+02000000014b421aa4622233390cdb87ea39e91760e69eeb10db4222e31cf354536a3beefd0100000000fdffffff010000000000000000056a03feeeee00000000
+
+# No signing required; it’s witness-free segwit
+bitcoin-cli -regtest -rpcwallet=test sendrawtransaction 02000000014b421aa4622233390cdb87ea39e91760e69eeb10db4222e31cf354536a3beefd0100000000fdffffff010000000000000000056a03feeeee00000000
+8d092b61ef3c1a58c24915671b91fbc6a89962912264afabc071a4dbfd1a484e
+
+```
+
+## User Stories
+
+Moving on from the more general release notes level feature description, we will
+describe a few common wallet patterns and how these can benefit from these
+updates, with or without the wallets making active changes.
+
+### Simple Payments
+
+If users wish to have more predictable RBF behavior, one way would be to opt-in
+to TRUC transactions. If adapted, wallets should only use confirmed outputs for
+TRUC transactions and construct them to stay within 10kvB (as opposed to the
+100kvB non-TRUC limit). This restricted limit still supports larger batch
+payments as well. If a wallet has no choice but to spend an unconfirmed input,
+if that input comes from a TRUC transaction, this new transaction would need to
+be below 1kvB.
```

Comment:
For this and all of the examples, I would suggest turning this into a bulleted list for clarity (presumably we use this to write code). Also, "if that input comes from a TRUC" isn't correct
```suggestion
to TRUC transactions. If adapted, wallets should:
- set version to 3
- only use confirmed outputs
- stay within 10kvB (as opposed to the 100kvB non-TRUC limit).
  - This restricted limit still supports larger batch payments.
  - If a wallet has no choice but to spend an unconfirmed input, the input must come from a TRUC transaction, and this new transaction needs to be below 1kvB.
```

Reply:
Took this